includes/payment-methods/class-wc-stripe-upe-payment-gateway.php

```
@@ -2149,6 +2152,195 @@ public function process_upe_redirect_payment( $order_id, $intent_id, $save_payme
+			// Hard failure: a real decline/error reported by the upstream payment provider.
+			$is_hard_failure = ! empty( $error_code ) && ! in_array( $error_code, $cancellation_codes, true );
+
+			if ( $is_hard_failure ) {
+				WC_Stripe_Logger::error(
+					'Checkout session redirect failed for order: ' . $order_id,
+					[
+						'session_status' => $status,
+						'error_code'     => $error_code,
+						'error_message'  => $error_message,
+					]
+				);
```

PHPMD (2.15.0)

[error] Avoid using static access to class 'WC_Stripe_Logger' in method 'process_checkout_session_redirect'. (undefined)

(StaticAccess)